### ISSUE TITLE
False positive "unresolved symbol" when referencing previous args in keyword arg defaults

### ISSUE THREAD

**Issue by chriskdon** [borkdude/clj-kondo#896]
**version**

`clj-kondo v2020.05.09`

**platform**

Native, OSX

**problem**

If you have a function with keyword arg defaults that reference previous args an "unresolved symbol" error is returned. 

**repro**

```clj
(ns repro)

(defn this-fails [x & {:keys [y] :or {y x}}]
  (prn "X" x "Y" y))
```

output from `clj-kondo --lint repro.clj`

```
repro.clj:3:41: error: unresolved symbol x
linting took 37ms, errors: 1, warnings: 0
```

**expected behavior**

The symbol `x` is defined, this should not be an error.

**Comment by borkdude** [comment]
Duplicate of #782.
